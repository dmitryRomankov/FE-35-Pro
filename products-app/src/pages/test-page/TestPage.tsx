import React from "react";

type TestPageProps = {
  name: string;
  isShow: boolean;
};

export const TestPage = ({ name, isShow }: TestPageProps) => {
  return (
    <div>
      <h1>Test page</h1>
      <h2>Hello {name}</h2>
      {isShow && <p>Description of the page</p>}
    </div>
  );
};
