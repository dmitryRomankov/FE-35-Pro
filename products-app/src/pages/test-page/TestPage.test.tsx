import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { TestPage } from "./TestPage";

describe("Test page", () => {
  test("Title renders correctly", () => {
    render(<TestPage name="Dmitry" isShow />);

    const headElement = screen.getByText(/test page/i);
    expect(headElement).toBeInTheDocument();
  });

  test("Description renders correctly", () => {
    render(<TestPage name="Dmitry" isShow />);

    const description = screen.getByText("Description of the page");
    expect(description).toBeInTheDocument();
  });

  test("Description should not be on the page", () => {
    render(<TestPage name="Dmitry" isShow={false} />);

    const description = screen.queryByText("Description of the page");
    expect(description).toBeNull();
  });

  test("Component renders correctly", () => {
    const tree = renderer
      .create(<TestPage name="Dmitry" isShow={false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
