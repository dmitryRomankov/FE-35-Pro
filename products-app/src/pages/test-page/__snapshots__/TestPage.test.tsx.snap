// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test page Component renders correctly 1`] = `
<div>
  <h1>
    Test page
  </h1>
  <h2>
    Hello 
    Dmitry
  </h2>
</div>
`;
